code/datums/traits/neutral.dm

```diff
@@ -97,3 +97,80 @@
 
 /datum/quirk/monochromatic/remove()
 	quirk_holder.remove_client_colour(/datum/client_colour/monochrome)
+
+/datum/status_effect/offering
+	id = "offering"
+	duration = -1
+	tick_interval = -1
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = null
+	/// The people who were offered this item at the start
+	var/list/possible_takers
+	/// The actual item being offered
+	var/obj/item/offered_item
+	/// The type of alert given to people when offered, in case you need to override some behavior (like for high-fives)
+	var/give_alert_type = /atom/movable/screen/alert/give
+
+/datum/status_effect/offering/on_creation(mob/living/new_owner, obj/item/offer, give_alert_override)
+	. = ..()
+	if(!.)
+		return
+	offered_item = offer
+	if(give_alert_override)
+		give_alert_type = give_alert_override
+
+	for(var/mob/living/carbon/possible_taker in orange(1, owner))
+		if(!owner.CanReach(possible_taker) || IS_DEAD_OR_INCAP(possible_taker) || !possible_taker.can_hold_items())
+			continue
+		register_candidate(possible_taker)
+
+	if(!possible_takers) // no one around
+		qdel(src)
+		return
+
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, .proc/check_owner_in_range)
+	RegisterSignal(offered_item, list(COMSIG_PARENT_QDELETING, COMSIG_ITEM_DROPPED), .proc/dropped_item)
+
+/datum/status_effect/offering/Destroy()
+	for(var/i in possible_takers)
+		var/mob/living/carbon/removed_taker = i
+		remove_candidate(removed_taker)
+	LAZYCLEARLIST(possible_takers)
+	return ..()
+
+/// Hook up the specified carbon mob to be offered the item in question, give them the alert and signals and all
+/datum/status_effect/offering/proc/register_candidate(mob/living/carbon/possible_candidate)
+	var/atom/movable/screen/alert/give/G = possible_candidate.throw_alert("[owner]", give_alert_type)
+	if(!G)
+		return
+	LAZYADD(possible_takers, possible_candidate)
+	RegisterSignal(possible_candidate, COMSIG_MOVABLE_MOVED, .proc/check_taker_in_range)
+	G.setup(possible_candidate, owner, offered_item)
```

Comment:
Should this register for qdel too to prevent hard deletes?

Reply:
I think status effects are cleared on mob del but I'd have to double check that